Sources/Utils.php

```diff
@@ -1727,11 +1727,12 @@ public static function checkMimeType(string $data, string $type_pattern, bool $i
 	 * Tries different modes to make files or directories writable.
 	 *
 	 * Wrapper function for PHP's chmod().
-	 *
+	 * 
 	 * @param string $path The full path of the file or directory.
+	 * @param array|string|null $chmod Specify a chmod, if left blank, SMF will auto pick the writable values.
 	 * @return bool Whether the file/dir exists and is now writable.
 	 */
-	public static function makeWritable(string $path): bool
+	public static function makeWritable(string $path, array|string|null $chmod = null): bool
```

Comment:
We have many places that call makeWritable and passing a second argument.  But it doesn't exist.  The code that called this was in Subs-Package.php.  Because it needs to be able to specify the chmod, I opted to fix it here by allowing us to pass it.  Both a string or array of chmod can be passed, then it will work.  However because the function returns true when writable, if your sending a chmod that should remove writable, make sure to expect that you will want a bool of false returned.